src/ui/SurveyUnit/Communication/CommunicationForm.tsx

```diff
@@ -0,0 +1,218 @@
+import Dialog from '@mui/material/Dialog';
+import { useMemo, useState } from 'react';
+import D from '../../../i18n/build-dictionary';
+import { surveyUnitIDBService } from '../../../utils/indexeddb/services/surveyUnit-idb-service';
+import {
+  communicationReasonEnum,
+  communicationStatusEnum,
+  communicationTypeEnum,
+} from '../../../utils/enum/CommunicationEnums';
+import CommunicationDialogContent from './CommunicationDialogContent';
+import CommunicationConfirmation from './CommunicationConfirmation';
+import { mediumRadioValues, reasonRadioValues, typeRadioValues } from '../../../utils/constants';
+
+enum Steps {
+  MEDIUM,
+  TYPE,
+  REASON,
+  VALIDATE,
+}
+
+namespace Steps {
+  export function after(value: Steps): Steps {
+    return value + 1;
+  }
+
+  export function before(value: Steps): Steps {
+    return value - 1;
+  }
+}
+
+export interface CommunicationRequestRadioData {
+  value: string;
+  label: string;
+  disabled: boolean;
+}
+
+export interface CommunicationRequestForm {
+  medium: string | undefined;
+  reason: string | undefined;
+  type: string | undefined;
+}
+
+interface CommunicationFormProps {
+  onClose: () => void;
+  surveyUnit: SurveyUnit;
+}
+
+/**
+ * Form to add a new contact attempt to a survey unit
+ *
+ * @param {() => void} onClose
+ * @param {SurveyUnit} surveyUnit
+ * @returns {TSX.Element}
+ */
+export function CommunicationForm({ onClose, surveyUnit }: Readonly<CommunicationFormProps>) {
+  const [step, setStep] = useState(Steps.MEDIUM);
+  const [communicationRequest, setCommunicationRequest] = useState({
+    medium: '',
+    reason: '',
+    type: '',
+  } as CommunicationRequestForm);
+
+  const setCommunicationRequestPropValue = (value: string) => {
+    const prop = Steps[step].toLocaleLowerCase();
+    setCommunicationRequest({ ...communicationRequest, [prop]: value });
+  };
+
+  let communicationTemplates = surveyUnit?.communicationTemplates ?? [];
+  const mediums = mediumRadioValues.map(m => {
```

Comment:
```suggestion
  const mediums: CommunicationRequestRadioData = mediumRadioValues.map(m => {
```

Why not setting the type right after the variable name instead of the `as`